updated data validation base code.

breadsp opened the pull request:
Updated data validation base code. 
Updated file-object code.  Added in new file for equipment/consumable/reagent/aliquot

breadsp committed:
updated data validation base code.  Updated file-object code.  Added in new file for equipment/consumable/reagent/aliquot

n2iw commented:
Biospecimen_validator.py
 Line 9: logic is bit hard to understand. Is "has_data_column.index[i[0]]" looking for index of index inside the list? Please take a look at following code, to see if it is what you meant to do. 
  ` for index, value in enumerate(has_data_column):`
     ` Biospecimen_object.valid_ID(header_name, value, pattern, valid_cbc_ids, error_msg, index, 'Error')`

n2iw commented:
demographic_data_validator.py
- Line 8: another way to write it is:`if 'Research_Participant_ID' in header_name:`

file_validator_object.py
- Line 144: when will `test_value != test_value`?
- Line 242: upload to S3 might fail due to network issue, better to use  try/except blocks to surround it

n2iw commented:
nci-seronet-data-validator.py
- Line 67, 68 and 138: Should use TEST_MODE instead of 'TestMode'.

There is an empty file "other_files_validator"

breadsp committed:
remove other_files

These files are not ready to use just yet

breadsp committed:
Updated and cleaned files

Tweaked code some to reduce duplicate function calls, updated looping logic per Ming's suggestion regarding indexes
